instagram-app/src/components/Authentication/withAuthenticate.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+const withAuthenticate = PostsPage => LoginPage =>
+  class extends React.Component {
+    state = {
+      loggedIn: false
+    };
+
+    componentDidMount() {
+      if (!localStorage.getItem('user')) {
+        this.setState({
+          loggedIn: false
+        });
+      } else {
+        this.setState({
+          loggedIn: true
+        });
+      }
+    }
+    componentDidUpdate(prevProps, prevState) {
+      if (prevState !== this.state) {
+      }
+    }
+
+    render() {
+      if (this.state.loggedIn) {
+        return <PostsPage />;
+      }
```

Comment:
To make this a bit cleaner you could also use a [Ternary Operator](https://developer.mozilla.org/enUS/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) here. To do that it would like something like this `this.state.loggedIn ? <PostsPage /> : <LoginPage />`